tracing/opentracing/interceptors_test.go

```diff
@@ -186,14 +190,42 @@ func (s *OpentracingSuite) assertTracesCreated(methodName string) (clientSpan *m
 }
 
 func (s *OpentracingSuite) TestPing_PropagatesTraces() {
-	ctx := s.createContextFromFakeHttpRequestParent(s.SimpleCtx(), true)
+	ctx := s.createContextFromFakeHttpRequestParent(s.SimpleCtx(), true, "")
 	_, err := s.Client.Ping(ctx, goodPing)
 	require.NoError(s.T(), err, "there must be not be an on a successful call")
 	s.assertTracesCreated("/mwitkow.testproto.TestService/Ping")
 }
 
+func (s *OpentracingSuite) TestPing_CustomOpName() {
+
+	const grpcMethod = "/CustomSvc/Name"
+
+	t := func(method string) string {
+		switch method {
+		case grpcMethod:
+			return "myopname"
+		default:
+			return "NO_OP_NAME"
+		}
+	}
+
+	ctx := s.createContextFromFakeHttpRequestParent(s.SimpleCtx(), true, t(grpcMethod))
+	_, err := s.Client.Ping(ctx, goodPing)
+	require.NoError(s.T(), err, "there must be not be an error on a successful call")
+
+	for _, v := range s.mockTracer.FinishedSpans() {
+		switch v.OperationName {
+		case t(grpcMethod):
+			break
+		case "NO_OP_NAME":
+			err = errors.New("invalid operation name")
+		}
+	}
+	require.NoError(s.T(), err, "operation name should not be default")
+}
```

Comment:
This test will do fine, but I'm very confused by this logic. What do you think about this simplification? Maybe I'm misunderstanding something?
```suggestion
func (s *OpentracingSuite) TestPing_CustomOpName() {
	customOpName := "customOpName"
	
	ctx := s.createContextFromFakeHttpRequestParent(s.SimpleCtx(), true, customOpName)
	_, err := s.Client.Ping(ctx, goodPing)
	require.NoError(s.T(), err, "there must be not be an error on a successful call")

	for _, v := range s.mockTracer.FinishedSpans() {
		require.Equal(s.T(), customOpName, v.OperationName)
	}
}
```

Reply:
I was being over-cautious. Your version is clearer, I was intending to provide an example at the same time, but updating the docs would be best!

Reply:
I think each span needs to be evaluated. The ping to generate my trace with a custom name is being accompanied by the trace for the ping itself(?):

`spanId=44, parentId=999, sampled=true, name=customOpName`
`spanId=48, parentId=46, sampled=true, name=/mwitkow.testproto.TestService/Ping`

Suggested change:

```go
func (s *OpentracingSuite) TestPing_CustomOpName() {

	customOpName := "customOpName"

	ctx := s.createContextFromFakeHttpRequestParent(s.SimpleCtx(), true, customOpName)
	_, err := s.Client.Ping(ctx, goodPing)
	require.NoError(s.T(), err, "there must be not be an error on a successful call")

	spans := s.mockTracer.FinishedSpans()
	spanNames := make([]string, len(spans))
	for _, span := range spans {
		spanNames = append(spanNames, span.OperationName)
	}

	require.Contains(s.T(), spanNames, customOpName, "finished spans must contain the custom operation name")

}
```